Added status information to both hooks (#403)
clerk/clerk-docs: docs/references/react/use-sign-up.mdx

| `setSession()` | A function that sets the active session. |
| `signUp` | An object that contains the current sign-up attempt status and methods to create a new sign-up attempt. |

### Result Status

| Values | Descriptiom |
| --- | --- |

> | `complete` | User has been created and custom flow can proceed to `setActive()` to create session. |

Reviewer: ```suggestion
| `complete` | The user has been created and custom flow can proceed to `setActive()` to create session. |
```

> | `abandoned` | The SignUp attempt will be abandoned if it was started more than 24 hours previously. |

Reviewer: ```suggestion
| `abandoned` | The sign-up attempt will be abandoned if it was started more than 24 hours previously. |
```

> | `missing_requirements` | A requirement from the [Email, Phone, Username](https://dashboard.clerk.com/last-active?path=user-authentication/email-phone-username) settings is missing. EG, in your Dashboard the Password setting is required but a password as not provided in the custom flow. |

Reviewer: ```suggestion
| `missing_requirements` | A requirement from the [Email, Phone, Username](https://dashboard.clerk.com/last-active?path=user-authentication/email-phone-username) settings is missing. For example, in the Clerk Dashboard, the Password setting is required but a password was not provided in the custom flow. |
```
